packages/build/src/utils/json.ts

```diff
@@ -9,12 +8,11 @@ export type RootPackageJson = { name: string; version: string }
 
 const ROOT_PACKAGE_JSON_PATH = fileURLToPath(new URL('../../package.json', import.meta.url))
 
-// TODO: Replace with dynamic `import()` once it is supported without
-// experimental flags
 export const importJsonFile = async function (filePath: string): Promise<PackageJson> {
-  const fileContents = await readFile(filePath, 'utf-8')
+  const fileUrl = filePath.startsWith('file://') ? filePath : `file://${filePath}`
 
-  return JSON.parse(fileContents)
+  const module = (await import(fileUrl, { assert: { type: 'json' } })) as { default: PackageJson }
```

Comment:
The `assert` keyword [has been deprecated](https://v8.dev/features/import-attributes#deprecation-and-eventual-removal-of-assert), so this won't work in Node 22+. 

And even before that we'd get a warning that we probably don't want:

```
(node:57010) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
```

The new keyword is `with`, but sadly that doesn't work with v18.14.0, which is [the lowest version we have to support](https://github.com/netlify/build/blob/f719b0719ad7b719211904f573d8f3539b58ecbb/packages/build/package.json#L157).

We should keep the `readFile` approach for now.